sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamCalcRel.java

```diff
@@ -256,18 +283,25 @@ public void setup() {
     }
 
     @ProcessElement
-    public void processElement(@FieldAccess("row") Row row, OutputReceiver<Row> r) {
+    public void processElement(
+        @FieldAccess("row") Row row, OutputReceiver<Row> r, ProcessContext c) {
```

Comment:
Haven't done a full review yet. Using `ProcessContext` creates an implicit access to the raw input row, which breaks our ability to pushdown projects outside of SQL. (Tests are going in shortly with #15886.) You'll need to use `MultiOutputReceiver` instead.